pkg/recipes/terraform/config/config_test.go

```diff
@@ -614,8 +616,21 @@ func Test_AddProviders(t *testing.T) {
 			err = json.Unmarshal(expectedConfigBytes, &expectedConfig)
 			require.NoError(t, err)
 
-			// This performs a deep comparison of the two maps.
-			require.Equal(t, expectedConfig, actualConfig)
+			if tc.checkProviderKeys {
+				// The AddProviders function accepts a map of UCPConfigured providers. As maps in Go do not guarantee iteration order, the sequence of providers in the resulting configuration can vary.
```

Comment:
If we sorted the two maps by the same condition, would we be able to guarantee that the iteration order is the same? i.e. sort by the name of the provider or something? Or would this change what values we want to verify?

Reply:
I started writing a long explanation but easier to discuss in mtg. i had thought about this. 
short ans: sorting in the test function doesn't help and updating the target function for this did not make sense for the intended functionality.